ouroboros-consensus/docs/GenesisDesignOverview.md

```diff
@@ -0,0 +1,404 @@
+
+# Eclipse detection
+
+That description of constitutes the expected path through the Genesis state
+machine for a (re)joining node, but there are other paths as well. If at any
+moment we detect an eclipse (see below), we transition back to Connecting as an
+attempt to escape. (TODO Maybe we need to check for eclipse before transition
+to CaughtUp so that a node that is fully eclipsed won't only try to escape once
+ever and then just give up?)
+
+The rule that triggers this transition has non-trivial consequences.
+
+## Total versus active stake
+
+There's an important distinction to start with.
+
+> _Total stake_ is all stake registered on the chain. _Active stake_ is the
+  stake held by CaughtUp block-producing nodes.
+
+The Shelley leadership schedule randomly distributes leadership opportunities
+according to total stake, such that there's an active slot (ie slot with >0
+leaders) on average once per 20 slots (ie a Geometric distribution). But only
+the holders of active stake will actually mint blocks in the slots they lead.
+(Technically, a node could mint a "late" block sometime after its labeled slot,
+but honest nodes aren't supposed to do that.)
+
+The underlying Ouroboros assumption is that, at all times, the _active_ stake
+is dominated by the honest stakeholders (ie the honest stakeholders, with
+properly configured nodes, that remain in CaughtUp enough of the time, and so
+on). If that's always true, then (prefixes of) the honest chain will win every
+density comparison forever.
+
+## Detection scenarios
+
+If I'm eclipsed, then I won't see (prefixes of) the honest chain. How can I
+notice? We consider two cases.
+
+* For a fresh node that's joining the network for the first time -- one that
+  was never before CaughtUp -- the only hope for detecting eclipse is for the
+  chain you're fed to be objectively too sparse. EG a chain that has blocks in
+  less than 40% of the active slots that that chain's ledger states' leadership
+  schedules identify as active. But this rule would backfire catastrophically
+  if Cardano mainnet ever reached (near) that threshold, thus this rule would
+  be introducing a new failure mode if enough honest stakeholders gave up
+  Cardano (without unregistering their stake) even in the absence of an
+  adversary.
+
+* For a node that's rejoining the network, there's another option, since it
+  already has selected a prefix of the (presumably) honest chain. If we were to
+  become eclipsed by an adversary, which can never have more than half of the
+  active stake, then -- given a "long enough" slot interval on either side --
+  we should see a portion of our recent chain when the density plummeted.
+
+We could instead collapse those two by defining the density up to the genesis
+block to be two times whatever the absolute cutoff is. So I'll only discuss the
+rejoining case from now on.
+
+## Balancing false positives and missed alarms
```

Comment:
Good document overall. Reminder to include all outputs from discussion with Alexander Russell in this section...